proparse/src/main/java/org/prorefactor/proparse/antlr4/PostLexer.java

```diff
@@ -78,25 +83,26 @@ public Token nextToken() throws TokenStreamException {
 
         }
       }
-    } catch (IOException | RecognitionException caught) {
-      throw new TokenStreamException(caught);
+    } catch (IOException | RecognitionException | TokenStreamException caught) {
+      throw new RuntimeException(caught);
     }
   }
 
   private ProToken defined() throws IOException {
+    LOGGER.trace("Entering defined()");
     // Progress DEFINED() returns a single digit: 0,1,2, or 3.
     // The text between the parens can be pretty arbitrary, and can
     // have embedded comments, so this calls a specific lexer function for it.
     getNextToken();
-    if (currToken.getType() == ProParserTokenTypes.WS)
+    if (currToken.getType() == PreprocessorParser.WS)
       getNextToken();
-    if (currToken.getType() != ProParserTokenTypes.LEFTPAREN)
+    if (currToken.getType() != PreprocessorParser.LEFTPAREN)
       throwMessage("Bad DEFINED function in &IF preprocessor condition");
     ProToken argToken = lexer.getAmpIfDefArg();
     getNextToken();
-    if (currToken.getType() != ProParserTokenTypes.RIGHTPAREN)
+    if (currToken.getType() != PreprocessorParser.RIGHTPAREN)
       throwMessage("Bad DEFINED function in &IF preprocessor condition");
-    return new ProToken(filenameList, ProParserTokenTypes.NUMBER, prepro.defined(argToken.getText().trim().toLowerCase()));
+    return new ProToken(PreprocessorParser.NUMBER, prepro.defined(argToken.getText().trim().toLowerCase()));
   }
 
   private void getNextToken() throws IOException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)